src/config/provider_instances.py

```
@@ -0,0 +1,244 @@
+    provider_cfg: Mapping[str, Any],
+    *,
+    file_field: str,
+    env_field: str,
+    inline_field: str,
+    legacy_env_names: Iterable[str] = (),
+) -> str:
+    file_path = str(provider_cfg.get(file_field) or "").strip()
+    if file_path:
+        try:
+            return read_secret_file(file_path)
+        except Exception:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)